SOLUTION_2.md

```diff
@@ -0,0 +1,109 @@
+# How to run
+
+First create a `.env` file following the `.env.example` as example. Set the environment variables according to your environment.
+Then run `docker-compose up -d`, this will build the images and start the containers, the database will be daemonized and will be waiting for connections. The node image will install dependencies and will terminate. Its default container command will display the available commands and their options.
+
+After that the easiest way to run multiple commands is to run the container with some terminal, for example:
+
+```
+  docker-compose run node bash
+```
+
+Then you are ready to start sending commands to the api, for exemple:
+
+```
+  npm run test load
+```
+
+To pass options to the underlyin command it is necessary to add a double-dash (`--`) after the command name:
+
+```
+  npm run test load -- -u henrique
+```
+
+# Challenge:
+
+## Improve the database calls to allow the program to be run any number of times (without complaining that the table already exists) (what would be a more permanent solution to this?);
+
+I did this by using docker and saving the queries to create the needed tables on the /docker-entrypoint-initdb.d folder, this
+guarantees that these queries will run only once (the first time the container is executed), however if any errors happen during their execution,
+the container will be restarted and since the postgres/data folder will not be empty, the scripts will not run again, meaning the database will
+(probably) be broken. To solve this, the problematic script must be fixed, the data folder emptied and the container re-created.
+
+A more permanent solution would be to use a database migration management lib. Then we would add a command to update the database with any new
+migrations when its container starts.
+
+## Improve the program to take a command-line argument with the name of the GitHub user;
+
+By using a command line parser lib I added commands to the api, `load` and `list`.
+By running `load` and passing it the `--user` (`-u` for short) option followed by the username, the caller can decide which user will be saved
+to the database.
+
+## Add more fields (to the database and to the API calls);
+
+The following filds where added to the `github_user` table:
+
```

Comment:
Hello @henriquerdd , sorry for the delay on our review!
Why the 191 char limit?

Reply:
Não tinha real necessidade, e na verdade coloquei no automático! Isso foi um costume que eu adquiri por conta desse problema: https://www.grouparoo.com/blog/varchar-191